api-call.py

```diff
@@ -1,29 +1,226 @@
+                                       url_obj['parameters'],
+                                       url_obj['header'])
+    academic_year = terms_data_response['data'][0]['attributes']['calendarYear']
+    term = terms_data_response['data'][0]['attributes']['season']
+    url_obj['textbooks_parameters'] = {'academicYear': academic_year,
+                                       'term': term,
+                                       'subject': 'CS',
+                                       'courseNumber': '161'}
+
+
+def get_stops_vehicles_on_route(url_obj):
+    """Get stops and vehicles on a given route
+    and determine where it is heading in real time
+    and the ETA for the stop
+    """
+
+    routes_data_response = get_api_data(url_obj['url_routes']+'/'+url_obj['route_id'],
+                                        url_obj['parameters'],
+                                        url_obj['header'])
+    route_name = routes_data_response['data']["attributes"]["description"]
+    stops = routes_data_response['data']["attributes"]["stops"]
+
+    for stop in stops:
+        stop_id = stop["stopID"]
+        description = stop["description"]
+        url_obj['parameters'] = {'stopID': stop_id, 'routeID': url_obj['route_id']}
+        arrivals_data_response = get_api_data(url_obj['url_arrivals'],
+                                              url_obj['parameters'],
+                                              url_obj['header'])
+        get_vehicle_id = arrivals_data_response['data'][0]['attributes']['arrivals'][0]['vehicleID']
+        get_eta_at_Stop = arrivals_data_response['data'][0]['attributes']['arrivals'][0]['eta']
+        vehicles_data_response = get_api_data(url_obj['url_vehicles']+'/'+get_vehicle_id,
+                                              {},
+                                              url_obj['header'])
+        get_vehicle_name = vehicles_data_response['data']['attributes']['name']
+        get_vehicle_heading = vehicles_data_response['data']['attributes']['heading']
 
+        print(f"Route ID: {url_obj['route_id']}, Route Name: {route_name}, Stop ID: {stop_id}, Stop Name: {description}, Vehicle Name: {get_vehicle_name}  , Vehicle Number: {get_vehicle_id}, Heading: {get_vehicle_heading}, ETA for arrival to Stop: {get_eta_at_Stop}")
 
-if __name__ == "__main__":
-    parameters = {
-            
-        }
 
-    header={"Content-Type": "application/x-www-form-urlencoded;charset=utf-8" , "Authorization":"Bearer CUTOuxWmuRfKQKynBGjFGmBROM9R"}
-        
-    api_call =  get_api_data("https://api.oregonstate.edu/v1/beaverbus/routes", parameters, header)
+if __name__ == '__main__':
+    """Driver function
+    """
+    while True:
+        show_tasks()
+        user_choice = get_user_choice()
+        url_obj = get_url(user_choice)
+        if user_choice == '0':
+            print(url_obj['url'])
+            break
+        if (
+            user_choice == '1'
+            or user_choice == '2'
```

Comment:
You can use `in` to improve the code here.

Reply:
done.